src/main/scala/mesosphere/marathon/core/task/state/MarathonTaskStatus.scala

```diff
@@ -0,0 +1,100 @@
+package mesosphere.marathon.core.task.state
+
+import org.apache.mesos
+
+// TODO ju handle garbage flag
+sealed trait MarathonTaskStatus {
+  def toMesosStateName: String = "TASK_" + toString.toUpperCase
+}
+
+object MarathonTaskStatus {
+  import org.apache.mesos.Protos.TaskState._
+
+  def fromMesosStateName(mesosStateName: String): Option[MarathonTaskStatus] = all.collectFirst {
+    case status: MarathonTaskStatus if status.toMesosStateName == mesosStateName => status
+  }
+
+  // If we're disconnected at the time of a TASK_LOST event, we will only get the update during
+  // a reconciliation. In that case, the specific reason will be shadowed by REASON_RECONCILIATION.
+  // Since we don't know the original reason, we need to assume that the task might come back.
+  val MightComeBack: Set[mesos.Protos.TaskStatus.Reason] = Set(
+    mesos.Protos.TaskStatus.Reason.REASON_RECONCILIATION,
+    mesos.Protos.TaskStatus.Reason.REASON_SLAVE_DISCONNECTED,
+    mesos.Protos.TaskStatus.Reason.REASON_SLAVE_REMOVED
+  )
+
+  val WontComeBack: Set[mesos.Protos.TaskStatus.Reason] = {
+    mesos.Protos.TaskStatus.Reason.values().toSet.diff(MightComeBack)
+  }
+
+  trait Terminal
+
+  //scalastyle:off cyclomatic.complexity
+  def apply(taskStatus: mesos.Protos.TaskStatus): MarathonTaskStatus = {
+    taskStatus.getState match {
```

Comment:
The conversion from Mesos TaskStatus to MarathonTaskStatus in currently done in the `TaskStatusUpdateProcessorImpl`, called by the `MarathonScheduler`. From a boundary perspective, I can imagine putting the conversion login into that layer, i.e. convert within the `MarathonScheduler` (or whatever will be that layer in the future) and only pass the MarathonTaskStatus into the `TaskStatusUpdateProcessor`. That's currently not easily possible because the conversion is lossy (health info's are dropped) and we still depend on the mesos TaskStatus elsewhere (yet we should encompass those values in a future Status).

I don't think we can use an existing enum as a default case without affecting the scheduling logic. I'd rather make this an `Option` or `Try[MarathonTaskStatus]` and fail fast if we receive something that's not mapped. Plus, write a test that makes sure that all `mesos.Protos.TaskState.values()` are mapped to a `MarathonTaskStatus`.